src/main/java/com/synopsys/integration/detect/lifecycle/run/RunManager.java

```diff
@@ -445,6 +468,52 @@ private void runBlackDuckProduct(ProductRunData productRunData, DetectConfigurat
+        logger.info("Will include the signature scanner tool.");
+        BlackDuckSignatureScannerOptions blackDuckSignatureScannerOptions = detectConfigurationFactory.createBlackDuckSignatureScannerOptions();
+        BlackDuckSignatureScannerTool blackDuckSignatureScannerTool = new BlackDuckSignatureScannerTool(blackDuckSignatureScannerOptions, detectContext);
+        SignatureScannerToolResult signatureScannerToolResult = blackDuckSignatureScannerTool.runScanTool(blackDuckRunData, projectNameVersion, dockerTargetData);
+        if (signatureScannerToolResult.getResult() == Result.SUCCESS && signatureScannerToolResult.getCreationData().isPresent()) {
+            codeLocationAccumulator.addWaitableCodeLocation(signatureScannerToolResult.getCreationData().get());
+        } else if (signatureScannerToolResult.getResult() != Result.SUCCESS) {
+            eventSystem.publishEvent(Event.StatusSummary, new Status("SIGNATURE_SCAN", StatusType.FAILURE));
+            eventSystem.publishEvent(Event.Issue, new DetectIssue(DetectIssueType.SIGNATURE_SCANNER, Arrays.asList(signatureScannerToolResult.getResult().toString())));
+        }
+        logger.info("Signature scanner actions finished.");
+    }
+
+    // When running the binary scanner on a non-docker target
+    private void runBinaryScanner(DetectConfigurationFactory detectConfigurationFactory, EventSystem eventSystem, CodeLocationNameManager codeLocationNameManager, DirectoryManager directoryManager, NameVersion projectNameVersion, CodeLocationAccumulator codeLocationAccumulator)
+        throws DetectUserFriendlyException {
+        runBinaryScanner(detectConfigurationFactory, eventSystem, codeLocationNameManager, directoryManager, projectNameVersion, DockerTargetData.NO_DOCKER_TARGET, codeLocationAccumulator);
+    }
+
+    private void runBinaryScanner(DetectConfigurationFactory detectConfigurationFactory, EventSystem eventSystem, CodeLocationNameManager codeLocationNameManager, DirectoryManager directoryManager, NameVersion projectNameVersion, DockerTargetData dockerTargetData, CodeLocationAccumulator codeLocationAccumulator)
+        throws DetectUserFriendlyException {
+        logger.info("Will include the binary scanner tool.");
+        if (null != blackDuckServicesFactory) {
+            BinaryScanOptions binaryScanOptions = detectConfigurationFactory.createBinaryScanOptions();
+            BlackDuckBinaryScannerTool blackDuckBinaryScanner = new BlackDuckBinaryScannerTool(eventSystem, codeLocationNameManager, directoryManager, new SimpleFileFinder(), binaryScanOptions, blackDuckServicesFactory);
+            Map<File, NameVersion> binaryScanPaths = blackDuckBinaryScanner.determineBinaryScanPaths(dockerTargetData, projectNameVersion);
+            if (binaryScanPaths.size() > 0) {
+                List<BinaryScanToolResult> results = blackDuckBinaryScanner.performBinaryScanActions(binaryScanPaths);
+                for (BinaryScanToolResult result : results) {
```

Comment:
We shouldn't need to iterate the results in the RunManager, some kind of BinaryScansResult with anyFailed() and getAllCodeWaitData() seems more appropriate.